boards/p1am_100/examples/pwm.rs

```diff
@@ -0,0 +1,47 @@
+#![no_std]
+#![no_main]
+
+extern crate cortex_m_rt;
+extern crate p1am_100 as hal;
+extern crate panic_halt;
+
+use hal::clock::GenericClockController;
+use hal::delay::Delay;
+use hal::pac::{CorePeripherals, Peripherals};
+use hal::prelude::*;
+use hal::pwm::Pwm4;
+
+use cortex_m_rt::entry;
+
+#[entry]
+fn main() -> ! {
+    let mut peripherals = Peripherals::take().unwrap();
+    let core = CorePeripherals::take().unwrap();
+
+    let mut clocks = GenericClockController::with_external_32kosc(
+        peripherals.GCLK,
+        &mut peripherals.PM,
+        &mut peripherals.SYSCTRL,
+        &mut peripherals.NVMCTRL,
+    );
+    let mut delay = Delay::new(core.SYST, &mut clocks);
+    let mut pins = hal::Pins::new(peripherals.PORT);
+
+    let _led = pins.led.into_mode::<hal::gpio::v2::AlternateE>();
```

Comment:
Is this leftover from something else? I don't think it makes sense for an LED to use peripheral function E.

Reply:
My intent was to use PWM to throb the LED between different brightnesses. I believe that's function E?

Reply:
Ah, I think I remember being confused by that in the HAL as well. One of the two (thumbv6 or thumbv7) does not take ownership of the pin, even though it needs it for PWM. That API could probably be improved.